src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -3612,5 +3612,235 @@ void N()
                 TypeParameter("T"),
                 Keyword("notnull"));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task DiscardVariableDeclaration()
+        {
+            await TestAsync(@"
+class X
+{
+    void N()
+    {
+        var _ = int.Parse("""");
```

Comment:
In this test `_` is a local named "_". Looks like we may need to check symbol information to make the correct determination.

Reply:
For instance this is valid
```csharp
var _ = 5;
_ = 6;
Console.WriteLine(_.ToString("X"));
```

Reply:
My bad, i wasn't aware of the fact that the `_` would be treated as a local here.
In that case i guess i can remove the VariableDeclarator case again with the same reasoning Cyrus gave above for the case of an unused parameter named `_`: From the compilers view this is not a discard therefore we wouldn't classify it as such.